debezium-connector-mysql/src/test/java/io/debezium/connector/mysql/MySqlAntlrDdlParserTest.java

```diff
@@ -81,6 +81,24 @@ public void beforeEach() {
                 SchemaNameAdjuster.create(), new CustomConverterRegistry(null), SchemaBuilder.struct().build(), false, false);
     }
 
+    @Test
+    @FixFor("DBZ-4583")
+    public void shouldProcessLargeColumn() {
+        String ddl = "create table if not exists tbl_large_col(\n"
+                + "`id` bigint(20) NOT NULL AUTO_INCREMENT,\n"
+                + "c1 blob(4294967295) NOT NULL,\n"
+                + "PRIMARY KEY (`id`)\n"
+                + ")";
+        parser.parse(ddl, tables);
+        assertThat(((MySqlAntlrDdlParser) parser).getParsingExceptionsFromWalker().size()).isEqualTo(0);
+        assertThat(tables.size()).isEqualTo(1);
+
+        Table table = tables.forTable(null, null, "tbl_large_col");
+
+        assertThat(table.columnWithName("c1").typeName()).isEqualTo("BLOB");
+        assertThat(table.columnWithName("c1").length()).isEqualTo(Integer.MAX_VALUE);
```

Comment:
Can you add an assertion on the actual value? So as to verify that truncating aspect?

Reply:
@jpechane I can but please bear in mind that the memory test requirments can go either 2 GB or 4 GB higher.

Reply:
@jpechane, in fact I'd also be happy if you do a once off test locally, as long as we precisely describe semantics in the docs, based on that tests outcome. We also should clarify the changed behavior in the release notes: whereas such large column would have failed the connector before, it will now be truncated.

Reply:
Ok, I see. Out of curiosity, how would we receive such large value from the database? Will it be truncated already in the format we receive from the binlog reader?

Reply:
@gunnarmorling IMHO we will not as binlog read will fail on trying to allocate the byte array with the required size.

Reply:
@gunnarmorling Looking at inarylogclient code it operats with ints only too.